x-pack/test/functional/page_objects/search_playground_page.ts

```diff
@@ -58,24 +73,63 @@ export function SearchPlaygroundPageProvider({ getService }: FtrProviderContext)
-        await browser.refresh();
+      async createOpenAIConnector(proxy: LlmProxy) {
+        await testSubjects.click('.gen-ai-card');
+        await testSubjects.setValue('nameInput', 'myConnector');
+        await testSubjects.setValue('config.apiUrl-input', `http://localhost:${proxy.getPort()}`);
+        await testSubjects.setValue('secrets.apiKey-input', 'myApiKey');
+
+        await testSubjects.click('create-connector-flyout-save-btn');
+      },
+
+      async expectAddedConnectorCallout() {
+        await testSubjects.existOrFail('addedConnectorCallout');
         await testSubjects.missingOrFail('connectToLLMChatPanel');
       },
 
-      async expectToStartChatPage() {
-        expect(await testSubjects.isEnabled('startChatButton')).to.be(true);
-        await testSubjects.click('startChatButton');
-        await testSubjects.existOrFail('chatPage');
+      async expectHideGenAIPanelConnector() {
+        await testSubjects.missingOrFail('connectToLLMChatPanel');
       },
     },
     PlaygroundChatPage: {
-      async expectChatWorks() {
+      async expectChatWindowLoaded() {
+        expect(await testSubjects.isEnabled('editContextActionButton')).to.be(true);
+        expect(await testSubjects.isEnabled('viewQueryActionButton')).to.be(true);
+        expect(await testSubjects.isEnabled('viewCodeActionButton')).to.be(true);
+
+        expect(await testSubjects.isEnabled('regenerateActionButton')).to.be(false);
+        expect(await testSubjects.isEnabled('clearChatActionButton')).to.be(false);
+        expect(await testSubjects.isEnabled('sendQuestionButton')).to.be(false);
+
         await testSubjects.existOrFail('questionInput');
+        const model = await testSubjects.find('summarizationModelSelect');
+        const defaultModel = await model.getVisibleText();
+
+        expect(defaultModel).to.equal('OpenAI GPT-3.5 Turbo');
+        expect(defaultModel).not.to.be.empty();
+
+        expect(
+          await (await testSubjects.find('manageConnectorsLink')).getAttribute('href')
+        ).to.contain('/app/management/insightsAndAlerting/triggersActionsConnectors/connectors/');
+
+        await testSubjects.click('sourcesAccordion');
+
+        expect(await testSubjects.findAll('indicesInAccordian')).to.have.length(1);
+      },
+
+      async sendQuestion() {
         await testSubjects.setValue('questionInput', 'test question');
         await testSubjects.click('sendQuestionButton');
-        await testSubjects.existOrFail('userMessage');
+      },
+
+      async expectChatWorks() {
+        const userMessageElement = await testSubjects.find('userMessage');
+        const userMessage = await userMessageElement.getVisibleText();
+        expect(userMessage).to.contain('test question');
+
+        const assistantMessageElement = await testSubjects.find('assistant-message');
+        const assistantMessage = await assistantMessageElement.getVisibleText();
+        expect(assistantMessage).to.contain('My response');
```

Comment:
Have you been able to check if we get a specific number of tokens in the message and specific number of retrievals docs?
And if yes, maybe let's write another test( if it's not covered in QA scenarios doc) where we open retrieved flyout and check what exactly we see there

Reply:
I have added an `expect` for the token tool tips. The document flyout and token tooltip tests will be part of the medium-priority tests.